quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/services/resources/DebugHostnameSettingsResource.java

```diff
@@ -117,10 +130,45 @@ public String debug(final @PathParam("realmName") String realmName) throws IOExc
     @GET
     @Path("/{realmName}/" + DEFAULT_PATH_SUFFIX + "/" + PATH_FOR_TEST_CORS_IN_HEADERS)
     @Produces(MediaType.TEXT_PLAIN)
-    public Response test(final @PathParam("realmName") String realmName) {
-        Response.ResponseBuilder builder = Response.ok(PATH_FOR_TEST_CORS_IN_HEADERS + "-OK");
-        String origin = keycloakSession.getContext().getRequestHeaders().getHeaderString(Cors.ORIGIN_HEADER);
-        builder.header(Cors.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
+    public Response test(final @PathParam("realmName") String realmName, @DefaultValue("false") @QueryParam("frontEnd") boolean frontEnd) {
+        String text = "OK";
+        String corsOrigin = keycloakSession.getContext().getRequestHeaders().getHeaderString(Cors.ORIGIN_HEADER);
+        URI requestUri = keycloakSession.getContext().getUri().getRequestUri();
+        String requestOrigin = UriUtils.getOrigin(requestUri);
+        URI frontendUri = keycloakSession.getContext().getUri(UrlType.FRONTEND).getBaseUri();
+
+        if (frontEnd) {
+            boolean originMatches = requestOrigin.equals(UriUtils.getOrigin(frontendUri));
+            HttpHeaders requestHeaders = keycloakSession.getContext().getRequestHeaders();
+            boolean proxied = requestHeaders.getHeaderString("Forwarded") != null || requestHeaders.getHeaderString("X-Forwarded-For") != null;
```

Comment:
Should we check for other `X-Forwarded-*` headers too, or at least change it to `X-Forwarded-Host` as it's most relevant for this case?

Reply:
> do you want to consolidate those and switch to X-Forwarded-Host instead?

That'd probably make sense.

Reply:
But it's not a blocker, approving regardless, feel free to merge.

Reply:
Refined to use the headers we have defined in the constants.